### DISCUSSION TITLE
./run_tests error - Python 2

### DISCUSSION THREAD

**Discussion by Bala-1005** [idaholab/moose#16433] [Q&A Getting Started] [answered]
Hello everyone,

I was creating a new app on moose and was able to create the object file that I needed. But once I did the ./run_tests -j4 command, I got the following error.

Traceback (most recent call last):
  File "./run_tests", line 11, in <module>
    from TestHarness import TestHarness
  File "/home/gnanasbr/temp_projects/moose/python/TestHarness/__init__.py", line 10, in <module>
    from .TestHarness import TestHarness
  File "/home/gnanasbr/temp_projects/moose/python/TestHarness/TestHarness.py", line 17, in <module>
    from FactorySystem.Factory import Factory
  File "/home/gnanasbr/temp_projects/moose/python/FactorySystem/__init__.py", line 13, in <module>
    from .Parser import Parser
  File "/home/gnanasbr/temp_projects/moose/python/FactorySystem/Parser.py", line 17
    print('failed to import pyhit - try running "make hit" in the $MOOSE_DIR/test directory.', file=sys.stderr)
                                                                                                   ^
SyntaxError: invalid syntax

What seems to be the problem here?

Thanks,
Bala

**Comment by GiudGiud** [comment]
Hello

This is odd. Is it a Python2 error with print() ? Can you tell us what 'which python' returns ?

Guillaume

**Comment by Bala-1005** [comment]
which python returns 
/usr/bin/python.
which python
/usr/bin/python

Thanks,
Bala

**Comment by GiudGiud** [comment, marked as answer]
and what does python --version return?
This is not the python from conda

**Reply by Bala-1005** [reply]
Hello,
python --version returns Python 2.7.5

Thanks,
Bala

**Reply by GiudGiud** [reply]
Ok so the issue is that you are using your system's python, instead of the conda python. 
which means that both
- it wont work because python 2 is no longer supported
- all the python libraries which we had you install are not accessible to that python.

Did you restart your terminal since you installed conda? (is it initialized is what I'm trying to get to) If not, do that.

If you did and it still does not work:
-What does which python3 return?
-Can you add the path to the conda python to the beginning of $PATH and try again?

**Reply by Bala-1005** [reply]
I see. But I am using a terminal on an HPC. I have other apps that seemed to have worked before. I just wanted to create a new application. I have to talk to my system administrator about conda since I have no idea how to go about it.
 python3 gives the following result

Python 3.6.8 (default, Apr  2 2020, 13:34:55) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-39)] on linux
Type "help", "copyright", "credits" or "license" for more information.

Thanks,
Bala

**Reply by GiudGiud** [reply]
So how did you install moose? Did you just load a module installed by your HPC team?
Our default installation process involves using conda, which comes with its own python(3).

**Reply by Bala-1005** [reply]
I used the following webpage for installation on HPC a while ago. 
https://mooseframework.inl.gov/getting_started/installation/hpc_install_moose.html

Thanks,
Bala

**Reply by GiudGiud** [reply]
Ok did you make sure to have the Python 3.x Development libraries installed?

**Reply by GiudGiud** [reply]
If they are already installed on the cluster for Python 3, then you can use that. If not you will need to either ask your administrator to install them, or install your own python locally (using conda is fairly easy)

**Reply by fdkong** [reply]
You could use `module av` to check what python modules are available on HPC. And then you could `module load` a python3 package.  Most of HPC systems should have such a package

**Reply by Bala-1005** [reply]
Thank you Fande and Guillaume. I couldnt find any python module available on our HPC. I am in communication with our system administrator about it. And hopefully I should be able to sort it out. 

Thanks,
Bala
